pkg/kando/location.go

```diff
@@ -38,17 +53,119 @@ func newLocationCommand() *cobra.Command {
 	cmd.AddCommand(newLocationDeleteCommand())
 	cmd.PersistentFlags().StringP(pathFlagName, "s", "", "Specify a path suffix (optional)")
 	cmd.PersistentFlags().StringP(profileFlagName, "p", "", "Pass a Profile as a JSON string (required)")
-	_ = cmd.MarkFlagRequired(profileFlagName)
+	cmd.PersistentFlags().StringP(repositoryServerFlagName, "r", "", "Pass a Repository Server CR as a JSON string (required for kopia based blueprints)")
 	return cmd
 }
 
+// NewDataMover creates an instance of DataMover Interface and returns
+// the preferred DataMover as per the arguments passed in kando command
+func NewDataMover(dm *DataMoverArgs, outputName, snapJson string) datamover.DataMover {
+	switch dm.Type {
+	case DataMoverTypeProfile:
+		return datamover.NewProfileDataMover(dm.Profile, outputName, snapJson)
+	case DataMoverTypeRepositoryServer:
+		return datamover.NewRepositoryServerDataMover(dm.RepositoryServer, outputName, snapJson)
+	default:
+		return nil
+	}
+}
+
 func pathFlag(cmd *cobra.Command) string {
 	return cmd.Flag(pathFlagName).Value.String()
 }
 
+// validateCommandArgs makes sure that we are getting exactly
+// one of --profile or --repository-server flags
+func validateCommandArgs(cmd *cobra.Command) error {
+	profileFlag := cmd.Flags().Lookup(profileFlagName).Value.String()
+	repositoryServerFlag := cmd.Flags().Lookup(repositoryServerFlagName).Value.String()
+	if profileFlag != "" && repositoryServerFlag != "" {
+		return errors.New("Either --profile or --repository-server can be provided")
+	}
+	if profileFlag == "" && repositoryServerFlag == "" {
+		return errors.New("Please provide either --profile or --repository-server as per the datamover you want to use")
+	}
+	return nil
+}
+
+func dataMoverFromCMD(cmd *cobra.Command, flag string) (datamover.DataMover, error) {
+	var dm *DataMoverArgs
+	var err error
+
+	dataMoverType, err := dataMoverTypeFromCMD(cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	switch dataMoverType {
+	case DataMoverTypeProfile:
+		dm, err = dataMoverArgsForProfile(cmd)
+		if err != nil {
+			return nil, err
+		}
+	case DataMoverTypeRepositoryServer:
+		dm, err = dataMoverArgsForRepositoryServer(cmd)
+		if err != nil {
+			return nil, err
+		}
+	default:
+		return nil, errors.New("Please provide either --profile or --repository-server as per the datamover you want to use")
+	}
+
+	switch flag {
+	case kopiaSnapshotFlagName:
+		return NewDataMover(dm, "", kopiaSnapshotFlag(cmd)), nil
+	case outputNameFlagName:
+		return NewDataMover(dm, outputNameFlag(cmd), ""), nil
+	default:
+		return nil, errors.New("Please provide either --profile or --repository-server as per the datamover you want to use")
+	}
```

Comment:
```suggestion
	var outputValue string
	switch flag {
	case kopiaSnapshotFlagName:
		outputValue = kopiaSnapshotFlag(cmd)
	case outputNameFlagName:
		outputValue = outputNameFlag(cmd)
	default:
		return nil, errors.New("Please provide either --output-name or --kopia-snapshot required by the command you want to use")
	}
	return NewDataMover(dm, "", outputValue), nil
```

Reply:
The position of the arguments in NewDataMover, depends on which flag we get so this won't be working @pavannd1